main.go

```diff
@@ -507,29 +502,43 @@ func (c *controller) sync() {
 			continue
 		}
 
-		// If there's an increase in replicas we poll for the new replicas to be ready
-		if _, ok := c.replicas[hashring]; ok && c.replicas[hashring] < *sts.Spec.Replicas {
-			// Iterate over new replicas to wait until they are running
-			for i := c.replicas[hashring]; i < *sts.Spec.Replicas; i++ {
-				start := time.Now()
-				podName := fmt.Sprintf("%s-%d", sts.Name, i)
+		var endpoints []string
+		// Iterate over new replicas to wait until they are running
+		for i := 0; i < int(*sts.Spec.Replicas); i++ {
```

Comment:
This seems like the right place to use a WaitGroup (https://gobyexample.com/waitgroups) and use a global timeout so we wait maximum `t` seconds and not `n*t`?

Reply:
makes sense to parallelize and reduce the wait time.  The slice 'endpoints' would need to be shared and updated across the go routines thru a Channel would have impact on the level parallelism.

Reply:
What is the consensus on this are we going to use the waitGrouos?

Reply:
I would vote for not using `waitGroups` in this case. 

- Employing `waitGroups` reduce the overall  wait time (for pods to come up),  but require handle synchronize append()s to a shared data structure across go routines. 
- The 1 min. is the maximum time that receive-controller wait for a pod to come up, i.e. if a pod does come up earlier than 1 min., 1 min. wait time does not occur. 
- With `waitGroups` -  if the one of pods does take 1 min. time to come up, would negate any advantage of employing waitGroups.